### PR TITLE
[31744] Fix: Disable dragging on mobile devices

### PR DESCRIPTION
**Pull request by ingamai**
#### Changes
- Only hide cdkDragHandle on mobile devices (not on mobile screen sizes) and disable dragging on touch

See ticket: https://community.openproject.com/projects/openproject/work_packages/31744/activity

### CODE REVIEW

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/cb8f79a1b8f43508448da5d84c579b51653508b9/frontend/src/app/modules/grids/grid/grid.component.ts#L38">frontend/src/app/modules/grids/grid/grid.component.ts#L38</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: frontend/src/app/modules/grids/grid/grid.component.html [hunk 1]
```diff
@@ -19,12 +19,14 @@
     <div class="grid--area-content widget-box"
          cdkDrag
          (cdkDragStarted)="drag.start(area)"
-         (cdkDragDropped)="drag.drop()">
+         (cdkDragDropped)="drag.drop()"
+         [cdkDragDisabled]="browserDetector.isMobile">
```
Comment (HDinger) [inline frontend/src/app/modules/grids/grid/grid.component.html]: This is rather ugly code style. Please add a method in the component and use it here.

File: frontend/src/app/modules/grids/grid/grid.component.html [hunk 2]
```diff
@@ -19,12 +19,14 @@
...
 
       <span *ngIf="drag.isDraggable"
             class="grid--area-drag-handle
                    icon
-                   icon-drag-handle hidden-for-mobile"
+                   icon-drag-handle"
+            [hidden]="browserDetector.isMobile"
```
Comment (HDinger) [inline frontend/src/app/modules/grids/grid/grid.component.html]: Why this change? People who use a really small window for their browser, see the drag handle but can't actually drag. If the `hidden-for-mobile` class is not enough, you should still use it in addition to the `[hidden]` attribute.

**Review by HDinger — COMMENTED** [review]
Works correctly. 👍 Please have a look at my comments.

File: frontend/src/app/modules/grids/grid/grid.component.html (hunk 2 above)
Comment (ingamai) [reply frontend/src/app/modules/grids/grid/grid.component.html]: I made this change because the handle should be shown, if dragging is possible. Due to the change you can also drag and drop on small screens, but only on desktop, not mobile devices. The class "hidden-for-mobile" would always hide the handle for small screens even if dragging would be enabled.

File: frontend/src/app/modules/grids/grid/grid.component.html (hunk 1 above)
Comment (ingamai) [reply frontend/src/app/modules/grids/grid/grid.component.html]: Okay, I changed it!

File: frontend/src/app/modules/grids/grid/grid.component.html (hunk 2 above)
Comment (HDinger) [reply frontend/src/app/modules/grids/grid/grid.component.html]: But it doesn't work on mobile sized browsers:
drag_mobile

And I guess it cannot work because of the changes we make to the grid on mobile screen size. Without having debugged it, I assume that the location in the grid cannot be determined correctly and therefore it does not work. I think pragmatism is needed here, so it's better to simply hide it on mobile screen size.

Comment (ingamai) [reply frontend/src/app/modules/grids/grid/grid.component.html]: Oh, you are right. I didn't remember the grid changes. I will add the class again.

**Review by HDinger — APPROVED** [review]
👍 Waiting for travis.
